### PR TITLE
Allow usage of consts and variables for stable metrics in static analysis

### LINKED ISSUE #81863: Metric validation static analysis should be able to support const strings for Subsystem and Name
Metric validation currently expects stable metrics to have strings directly in the metric definition. However, we should probably also allow consts and vars which are declared in the pkg/file scope, since those can easily be evaluated to their respective string values.

/assign @serathius 
/sig instrumentation

### PR DESCRIPTION
**Pull request by serathius**
/kind feature

**What this PR does / why we need it**:

Fixes #81863

```release-note
NONE
```
/assign @logicalhan

### CODE REVIEW

File: test/instrumentation/decode_metric.go [hunk 1]
```diff
@@ -130,10 +130,11 @@ func decodeLabels(expr ast.Expr) ([]string, error) {
 		if !ok {
 			return nil, newDecodeErrorf(bl, errLabels)
 		}
-		if bl.Kind != token.STRING {
+		value, err := stringValue(bl)
+		if err != nil {
 			return nil, newDecodeErrorf(bl, errLabels)
```
Comment (logicalhan) [inline test/instrumentation/decode_metric.go]: Is this error supposed to be `errLabels`? I would think you would actually want to use the error returned from `stringValue`.

File: test/instrumentation/decode_metric.go [hunk 2]
```diff
@@ -160,14 +161,30 @@ func (c *metricDecoder) decodeOpts(expr ast.Expr) (metric, error) {
...
-				return m, newDecodeErrorf(expr, errNonStringAttribute)
-			}
-			if k.Kind != token.STRING {
+			var value string
+			var err error
+			switch v := kv.Value.(type) {
+			case *ast.BasicLit:
+				value, err = stringValue(v)
+				if err != nil {
+					return m, newDecodeErrorf(expr, errNonStringAttribute)
```
Comment (logicalhan) [inline test/instrumentation/decode_metric.go]: Is there a specific reason you want to construct the error here from the expr as opposed to use the error returned from stringValue above?

Comment (logicalhan) [reply test/instrumentation/decode_metric.go]: ?

Comment (serathius) [reply test/instrumentation/decode_metric.go]: After rethinking I changed it to return error from `stringValue`

**Comment by serathius** [comment]
ping @logicalhan

**Review by logicalhan — APPROVED** [review]
/lgtm

**Comment by serathius** [comment]
@logicalhan can you approve?

**Pull request merged**